Src/Plugins/Security/Security.SharedStrings.txt

```diff
@@ -30,7 +30,7 @@
   $HockeyAppToken=(?si)hockey.{0,50}[`'"\s>]?(?<refine>[0-9a-f]{32})([^0-9a-f]|$)
   $MySqlConnectionStringAdo=(?i)Server\s*=(?-i)(?<host>[0-9a-z\-]{3,63}\.mysql\.database\.azure\.com.*)(,|;).*Uid=(?<account>[a-z\@\-]{1,120})(,|;).*Pwd=(?<password>[^;]{8,128});
   $PostgreSqlConnectionStringAdo=(?i)Host\s*=\s*(?-i)(?<host>[0-9a-z\-_\.]{3,91})((?i);Port\s*=\s*(?<port>[0-9]{1,5}))?;Database\s*=\s*(?-i)(?<database>[0-9a-z\-_]{3,91})(?i);Username\s*=\s*(?<account>[^,;]+);Password\s*=\s*(?<password>[^,;\s]+)
-  $SqlConnectionStringAdo=(?i)Server\s*=\s*tcp:(?<host>[^,;]+)(,|;).*(User ID|Uid)\s*=\s*(?<account>[^;]+);.*(Password|Pwd)\s*=\s*(?<password>[^;]+);
-  $SqlConnectionStringJdbc=(?i)jdbc:sqlserver:\/\/(?<host>[^:]+):.*;user\s*=\s*(?<account>[^@]+)@[^;]+;password\s*=\s*(?<password>[^;]+);
-  $SqlConnectionStringPhp=(?i)sqlsrv:server\s*=\s*tcp:(?<host>[^,]+),[^,]+,\s*"(?<account>[^"]+)",\s*"(?<password>[^"]+)"
+  $SqlConnectionStringAdo=(?i)(((server|data source)\s*=\s*[^;]+|(initial catalog|database)\s*=\s*[^;]+|(user id|uid)\s*=\s*[^;]+|(password|pwd)\s*=\s*[^;]+).+?){4}
```

Comment:
>$SqlConnectionStringAdo [](start = 2, length = 23)

these changes add database but don't solve the order-sensitivity issue (which is the name of this PR). we must be deferring that work? in other words, this regex will trim user data if it happens to follow the password data.

Reply:
in theory it is working order-insensitive. if u check, i added {4} based on the whole group. With that, it will look for 4 patterns that might contain server/database/user/password. when it finds 4, it will get a match

---
In reply to: [569699776](https://github.com/microsoft/sarif-pattern-matcher/pull/160#discussion_r569699776) [](ancestors = 569699776)

Reply:
what i didn't do yet is the same thing for jdbc and php...it's getting trickier since we have only one validator. so, i just fixed to get dabase for now

---
In reply to: [569700938](https://github.com/microsoft/sarif-pattern-matcher/pull/160#discussion_r569700938) [](ancestors = 569700938,569699776)